fast64_internal/oot/c_writer/oot_scene_table_c.py

```diff
@@ -1,75 +1,144 @@
+	isEndOfExistingScenes = False
+	# used to add the "// Debug-only scenes" comment after non-debug scenes
+	isEndOfNonDebugScenes = False
+
+	# add the actual lines with the same formatting
+	for i in range(len(data)):
+		# add a comment to show when it's new scenes
+		if isEndOfExistingScenes:
+			fileData += "// Added scenes\n"
+			isEndOfExistingScenes = False
+		
+		# adds the "// Debug-only scenes" comment after SCENE_GANON_TOU
+		if isEndOfNonDebugScenes:
+			fileData += debugLine
+			isEndOfNonDebugScenes = False
+
+		fileData += f"/* 0x{i:02X} */ DEFINE_SCENE("
+		fileData += ", ".join(str(d) for d in data[i])
+
+		if data[i][0] == "testroom_scene":
+			isEndOfExistingScenes = True
+		if data[i][0] == "ganon_tou_scene":
+			isEndOfNonDebugScenes = True
+
+		fileData += ")\n"
+
+	# return the string containing the file data to write
+	return fileData
 
 def modifySceneTable(scene, exportInfo):
+	'''Edit the scene table with the new data'''
 	exportPath = exportInfo.exportPath
-	sceneTable, sceneFileData, start, end = readSceneTable(exportPath)
-	entry = getSceneTableEntryBySceneName(sceneTable, scene.name if scene is not None else exportInfo.name)
-
-	if scene is not None:
-		if entry is None:
-			sceneTable.append(SceneTableEntry("UNTITLED_SCENE", scene.name + "_scene", None, '0', str(scene.sceneTableEntry.drawConfig), '0'))
-		else:
-			entry.sceneType = entry.sceneType
-			entry.config = str(scene.sceneTableEntry.drawConfig)
+	fileData, header, debugLine = getSceneTable(exportPath)
+	sceneName, sceneTitle, sceneID, sceneUnk10, sceneUnk12, sceneIndex = getSceneParams(scene, exportInfo)
+
+	if scene is None:
+		sceneDrawConfig = None
 	else:
-		if entry is not None:
-			sceneTable.remove(entry)
-	writeSceneTable(sceneTable, sceneFileData, start, end, exportPath)
\ No newline at end of file
+		sceneDrawConfig = scene.sceneTableEntry.drawConfig
+
+	sceneParams = [sceneName, sceneTitle, sceneID, sceneDrawConfig, sceneUnk10, sceneUnk12]
+
+	# check if it's a custom scene name
+	if sceneIndex is None: 
+		isCustom = True
+	else:
+		isCustom = False
+
+	# if so, check if the custom scene already exists in the data
+	# if it already exists set isCustom to False to consider it like a normal scene
+	if isCustom:
+		for i in range(len(fileData)):
+			if fileData[i][0] == scene.name.lower() + "_scene":
+				sceneIndex = i
+				isCustom = False
+				break
+
+	# edit the current data or append new one if we are in a ``Custom`` context
+	for i in range(6):
+		if isCustom == False and sceneParams[i] is not None and fileData[sceneIndex][i] != sceneParams[i]:
+			fileData[sceneIndex][i] = sceneParams[i]
+		elif isCustom:
+			fileData.append(sceneParams)
+			sceneIndex = len(fileData) - 1
+			break
+
+	# remove the scene data if scene is None (`Remove Scene` button)
+	try:
+		if scene is None:
+			fileData.remove(fileData[sceneIndex])
+	except:
+		raise PluginError("ERROR: Scene not found in ``scene_table.h``!")
```

Comment:
```suggestion
	if scene is None:
		if sceneIndex < len(fileData):
			fileData.pop(sceneIndex)
		else:
			raise PluginError("ERROR: Scene not found in ``scene_table.h``!")
```

Reply:
I had to add a check to prevent the removing bug you mentioned earlier and I'm not convinced by the way I added that suggestion